lta/picker.py

```diff
@@ -92,90 +102,71 @@ def _expected_config(self) -> Dict[str, Optional[str]]:
+        dest = tr["dest"]
+        path = tr["path"]
         # query the file catalog for the source files
-        self.logger.info(f"Asking the File Catalog about files in {src_split}:{src_path}")
+        self.logger.info(f"Asking the File Catalog about files in {source}:{path}")
         query_dict = {
             "locations.site": {
-                "$eq": f"{src_site}"
+                "$eq": source
             },
             "locations.path": {
-                "$regex": f"^{src_path}"
+                "$regex": f"^{path}"
             }
         }
         query_json = json.dumps(query_dict)
         fc_response = await fc_rc.request('GET', f'/api/files?query={query_json}')
-        self.logger.info(f'File Catalog returned {len(fc_response["files"])} file(s) to process.')
-        # TODO: If there are no files, that seems like something to alert on
-        # if len(fc_response["files"]) < 1:
-        #     self.alerter.error("TR {UUID} wanted to archive {site}:{path} but the File Catalog reported 0 files!")
-        #     lta_rc.request('PATCH', '/TransferRequest/{uuid}', {"quarantine": "file_catalog_count == 0"})
-        #     return
-        # for each file provided by the catalog
-        bulk_create: FileList = []
+        num_files = len(fc_response["files"])
+        self.logger.info(f'File Catalog returned {num_files} file(s) to process.')
+        # if we didn't get any files, this is bad mojo
+        if num_files < 1:
+            self.logger.info(f'There are no files to process for TransferRequest {tr["uuid"]}.')
+            quarantine = {
+                "status": "quarantined",
+                "reason": "File Catalog returned zero files for the TransferRequest",
+            }
+            await lta_rc.request('PATCH', f'/TransferRequests/{tr["uuid"]}', quarantine)
+            return
+        # query the file catalog for the full records
+        catalog_records = []
         for catalog_file in fc_response["files"]:
-            bulk_create.extend(await self._do_work_catalog_file(lta_rc, tr, fc_rc, dests, catalog_file))
-        # 3. Update the LTA DB with Files needed for bundling
-        self.logger.info(f'Identified {len(bulk_create)} transfer(s) to add to the LTA DB.')
+            catalog_record = await fc_rc.request('GET', f'/api/files/{catalog_file["uuid"]}')
+            catalog_records.append(catalog_record)
+        # add up the sizes of everything returned by the catalog
+        packing_list = []
+        total_size = 0
+        for catalog_record in catalog_records:
+            file_size = catalog_record["file_size"]
+            #                    0: size    1: full record
+            packing_list.append((file_size, catalog_record))
+            total_size += file_size
+        # divide that by the size requested at the destination
+        bundle_size = self.sites[dest]["bundle_size"]
+        num_bundles = math.floor((float(total_size) / float(bundle_size)) + 0.5)
```

Comment:
Wouldn't `math.ceil(total_size / bundle_size)` be simpler than a floor with +0.5?

Actually, running a test comparing the two, they give slightly different answers.  The current line is actually rounding the answer, and produces identical results to `round(total_size / bundle_size)`.  If this is what we want to do, using the explicit rounding function may be more explicit.

Also note that in python3 you don't need to explicitly cast to float for division.  `/` is float division, while `//` is int division.

Reply:
They are slightly different in that `round()` breaks ties toward the even number and `math.floor()` after `+0.5` breaks ties toward the next highest integer.
```>>> import math
>>> round(4.5)
4
>>> math.floor(4.5+0.5)
5
```

However, you're right. I agree that `round()` is more explicit; clearly communicating the intent of the code.

Ironically, I tried the `/` division experiment in the Python REPL. It was a Python 2 REPL and it was doing integer division. Having just tried it again in a Python 3 REPL, I confirm that the `float()` casts are not necessary. I've removed them to simplify the code.

The new version is much nicer. Thanks!

Reply:
Round does not in fact always round down:
```
>>> round(5.5)
6
```
The discrepancy is due to floating point precision.  4.5 is probably actually 4.49999...  The pretty printing is getting in the way of how it is at the byte level in IEEE format.

Reply:
Correct, `round()` rounds up and down, but always breaks ties _toward the even number_:
```
>>> round(1.5)
2
>>> round(2.5)
2
>>> round(3.5)
4
>>> round(4.5)
4
>>> round(5.5)
6
>>> round(6.5)
6
>>> round(7.5)
8
>>> round(8.5)
8
```

Reply:
Interesting.  Strange, but I'm not sure I care enough; and in real life calculations this is a very small edge case we probably don't care about (I want to see the sizes that divide to get exactly .5).